src/App.tsx

```diff
@@ -1,24 +1,214 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { Select } from './types/Select';
+import {
+  getTodos,
+  deleteTodo,
+  patchTodoCompleted,
+  patchTodoTitle,
+} from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+import { Errors } from './types/Errors';
+import { Header } from './components/Header';
 
-const USER_ID = 0;
+export const USER_ID = 9960;
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[] | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState(Select.ALL);
+  const [typeError, setTypeError] = useState<Errors | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [completedAll, setCompletedAll] = useState(false);
+  const [loaderTodo, setLoaderTodo] = useState(0);
+
+  const filteredList = useMemo(() => {
+    return todoList && todoList.filter((todo) => {
+      switch (selectedFilter) {
+        case Select.ACTIVE:
+          return !todo.completed;
+        case Select.COMPLETED:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+  }, [selectedFilter, todoList]);
+
+  const countItemLeft = todoList?.filter(todo => !todo.completed).length;
+
+  const deleteClickHandlerItem = (id: number) => {
+    if (id) {
```

Comment:
id is required param, so if statement is useless, also if id will be equal to 0 - this condition will be falsy